show/muxcable.py

```diff
@@ -0,0 +1,160 @@
+        if y_cable_asic_table is not None:
+            y_cable_asic_table_keys = y_cable_asic_table.getKeys()
+            if port in y_cable_asic_table_keys:
+                port_status_dict = {}
+                port_status_dict["mux_cable"] = {}
+                (status, fvs) = y_cable_asic_table.get(str(port))
+                if status is not True:
+                    click.echo("could not retrieve port values for port".format(port))
+                fvp = dict(fvs)
+                status_value = fvp.get("status", None)
+                port_status_dict["mux_cable"][port] = {"status": status_value}
+
+                click.echo("muxcable Ports status : \n {}".format(json.dumps(port_status_dict, indent=4)))
+            else:
+                click.echo("this is not a valid port present on mux_cable".format(port))
+        else:
+            click.echo("there is not a valid asic table for this asic_index".format(asic_index))
+
+    else:
+
+        port_status_dict = {}
+        port_status_dict["mux_cable"] = {}
+        for namespace in namespaces:
+            asic_id = multi_asic.get_asic_index_from_namespace(namespace)
+            for key in port_table_keys[asic_id]:
+                (status, fvs) = y_cable_tbl[asic_id].get(key)
+                if status is not True:
+                    click.echo("could not retrieve port values for port".format(key))
+                fvp = dict(fvs)
+                status_value = fvp.get("status", None)
+                port_status_dict["mux_cable"][key] = {"status": status_value}
+
+        click.echo("muxcable Ports status : \n {}".format(json.dumps(port_status_dict, indent=4)))
+
+
+@muxcable.command()
+def summary():
+    """Show muxcable summary information"""
+
+    state_db = {}
+    y_cable_tbl = {}
+    port_table_keys = {}
+
+    # Getting all front asic namespace and correspding config and state DB connector
+
+    namespaces = multi_asic.get_front_end_namespaces()
+    for namespace in namespaces:
+        asic_id = multi_asic.get_asic_index_from_namespace(namespace)
+        state_db[asic_id] = swsscommon.DBConnector("STATE_DB", REDIS_TIMEOUT_MSECS, True, namespace)
+        y_cable_tbl[asic_id] = swsscommon.Table(state_db[asic_id], "HW_MUX_CABLE_TABLE_NAME")
+        port_table_keys[asic_id] = y_cable_tbl[asic_id].getKeys()
+
+    port_status_dict = {}
+    port_status_dict["mux_cable"] = {}
+    for namespace in namespaces:
+        asic_id = multi_asic.get_asic_index_from_namespace(namespace)
+        for key in port_table_keys[asic_id]:
+            (status, fvs) = y_cable_tbl[asic_id].get(key)
+            if status is not True:
+                click.echo("could not retrieve port values for port".format(key))
+            fvp = dict(fvs)
+            status_value = fvp.get("status", None)
+            port_status_dict["mux_cable"][key] = {"status": status_value}
+
+    click.echo("muxcable Ports status : \n {}".format(json.dumps(port_status_dict, indent=4)))
```

Comment:
I think we should consider allowing a `--json` flag to output results in JSON format, otherwise output in a human-readable tabular format using `tabulate`.

Reply:
fixed